pkg/sidecar/config/config.go

```diff
@@ -173,13 +175,17 @@ func (s *ScyllaConfig) setupEntrypoint(ctx context.Context) (*exec.Cmd, error) {
 		devMode = "1"
 	}
 
+	shards, err := strconv.Atoi(options.GetSidecarOptions().CPU)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
 	args := []string{
 		fmt.Sprintf("--listen-address=%s", m.IP),
 		fmt.Sprintf("--broadcast-address=%s", m.StaticIP),
 		fmt.Sprintf("--broadcast-rpc-address=%s", m.StaticIP),
 		fmt.Sprintf("--seeds=%s", strings.Join(seeds, ",")),
 		fmt.Sprintf("--developer-mode=%s", devMode),
-		fmt.Sprintf("--smp=%s", options.GetSidecarOptions().CPU),
+		fmt.Sprintf("--smp=%d", shards),
```

Comment:
To be on the safe side I'd set that only if cpuset is disabled.

Reply:
That's not what I gathered from my discussions with @denesb . SMP refers to the number of shards and each cpu can serve more than one shard afaict.

Reply:
You want shard-per-core, this would be useful if we supported shard < available cores.

But you are correct it's safer to specify smp.

Reply:
> SMP refers to the number of shards and each cpu can serve more than one shard afaict.

Each CPU can host exactly one shard only, but in some cases some CPUs are reserved for interrupt processing or other side-tasks. So number of CPUS >= number of shards.